src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -125,6 +129,10 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
         return null;
       case ClassComponent:
         transferOutput(workInProgress.child, workInProgress);
+        // We are leaving this subtree, so pop context if any.
+        if (isContextProvider(workInProgress)) {
+          popContextProvider();
```

Comment:
If we unwind to catch an error, we need to unwind the context as well so that we can continue with the next sibling after the error boundary.